src/sql/base/browser/ui/table/plugins/headerFilter.plugin.ts

```diff
@@ -0,0 +1,357 @@
+// Adopted and converted to typescript from https://github.com/danny-sg/slickgrid-spreadsheet-plugins/blob/master/ext.headerfilter.js
+// heavily modified
+import 'vs/css!sql/base/browser/ui/table/media/table';
+
+import { mixin } from 'vs/base/common/objects';
+import { SlickGrid } from 'angular2-slickgrid';
+import { Button } from '../../button/button';
+import { attachButtonStyler } from 'sql/common/theme/styler';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { indexOf } from 'vs/platform/files/common/files';
+
+export class HeaderFilter {
+
+	public onFilterApplied = new Slick.Event();
+	public onCommand = new Slick.Event();
+
+	private grid;
+	private handler = new Slick.EventHandler();
+	private defaults = {
+		filterImage: 'src/sql/media/icons/filter.svg',
+		sortAscImage: 'sort-asc.gif',
+		sortDescImage: 'sort-desc.gif'
+	};
+
+	private $menu;
+	private options: any;
+	private okButton: Button;
+	private clearButton: Button;
+	private cancelButton: Button;
+	private workingFilters: any;
+	private columnDef: any;
+
+	constructor(options: any, private _themeService: IThemeService) {
+		this.options = mixin(options, this.defaults, false);
+	}
+
+	public init(grid: Slick.Grid<any>): void {
+		this.grid = grid;
+		this.handler.subscribe(this.grid.onHeaderCellRendered, (e, args) => this.handleHeaderCellRendered(e , args))
+				.subscribe(this.grid.onBeforeHeaderCellDestroy, (e, args) => this.handleBeforeHeaderCellDestroy(e, args))
+				.subscribe(this.grid.onClick, (e) => this.handleBodyMouseDown)
+				.subscribe(this.grid.onColumnsResized, () => this.columnsResized());
+
+		this.grid.setColumns(this.grid.getColumns());
+
+		$(document.body).bind('mousedown', this.handleBodyMouseDown);
```

Comment:
this will also catch clicks into things like Object Explorer or Query Editor or etc.?

Reply:
Might be better to use the context view class here to handle a dropdown menu rather than jquery